### PR TITLE
cc dhm alt

### PR DESCRIPTION
**Pull request by NirEkhauz**
## Description
Add the 'MBEDTLS_DHM_ALT' #Define in order to support the alternate DHM.

## Status
**READY/IN DEVELOPMENT/HOLD**

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

### CODE REVIEW

File: include/mbedtls/dhm.h [hunk 1]
```diff
@@ -23,7 +23,13 @@
 #ifndef MBEDTLS_DHM_H
 #define MBEDTLS_DHM_H
 
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
 #include "bignum.h"
+#if !defined(MBEDTLS_DHM_ALT)
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: This guards the `extern "C" { ...` path [below](https://github.com/ARMmbed/mbedtls/pull/1239/files#diff-12e227586f72d7fc03d416921e47b43aR148) but not its closing `}`. I'd suggest to proceed as in [sha1.c](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/sha1.h).

File: include/mbedtls/config.h [hunk 2]
```diff
@@ -275,6 +275,8 @@
 //#define MBEDTLS_SHA1_ALT
 //#define MBEDTLS_SHA256_ALT
 //#define MBEDTLS_SHA512_ALT
+//#define MBEDTLS_DHM_ALT
```
Comment (hanno-arm) [inline include/mbedtls/config.h]: This should be placed so that the list of alternative options is in alphabetical order.

Comment (NirEkhauz) [reply include/mbedtls/config.h]: OK

File: include/mbedtls/dhm.h (hunk 1 above)
Comment (NirEkhauz) [reply include/mbedtls/dhm.h]: You are right!
Thanks.

**Comment by hanno-arm** [comment]
I'd prefer this PR to be rebased on the current `development` since recent merges have introduced conflicts with this PR. @yanesca, would you be ok with that?

**Comment by yanesca** [comment]
@hanno-arm I think our current policy is that after the review only merges are allowed, no rebase or cherry-pick.

@gilles-peskine-arm please correct me if I am wrong.

**Comment by gilles-peskine-arm** [comment]
@yanesca This isn't an absolute policy, I rather see it as a means to achieve an objective, but it's fine if we achieve the objective by other means. The objective is that everything that goes into the product gets reviewed formally, apart from ChangeLog entries and from trivial merges. By “reviewed formally”, I mean that 1. the review happens and 2. there's a trace that the review happened.

If a rebase happens on a PR, the modified must be re-reviewed in its new context, and that practically requires doing the review from scratch. The alternative is to review a merge commit. Reviewing a merge commit is not easy, but it is a clear objective. In contrast, reviewing a rebase is fuzzier: it's easy to check that issues that came up originally weren't forgotten, but it's a lot harder to be confident that no new issues arose. It's also more difficult to verify after the fact that the review was done correctly (i.e. that the reviewers did look at the final version of the code, and not just some older version). This is why I think that in general, it's better to merge development into a PR branch and review the merge commit, rather than rebase and re-review. But rebasing is ok **if everybody (author, reviewers, gatekeeper) is confident that the review was done properly and traceably**.

Cc @Patater who is more fond of rebasing than I am.

**Comment by NirEkhauz** [comment]
So bottom line,
Rebase or Merge?

**Comment by hanno-arm** [comment]
@NirEkhauz Could you please go forward with a rebase?

**Comment by sbutcher-arm** [comment]
@gilles-peskine-arm - LGTM. Can you please review and merge?

**Comment by gilles-peskine-arm** [comment]
CI failed only in a timing test on FreeBSD which is a known unrelated issue. Ok to merge.

**Pull request merged**